### PR TITLE
Maintain sets of possible values for each slot in item collection

### LINKED ISSUE #112: Maintain sets of possible values for each slot in ItemCollection

### PR DESCRIPTION
**Pull request by NoB0**
Implementation of `get_possible_property_values`.
Fixes #112

### CODE REVIEW

File: usersimcrs/user_modeling/preference_model.py [hunk 1]
```diff
@@ -143,7 +143,8 @@ def get_item_preference(self, item_id: str) -> float:
         return preference
 
     def _check_slot_exists(self, slot: str) -> None:
-        """Checks if slot exists in the domain and throws an exception if not."""
```
Comment (kbalog) [inline usersimcrs/user_modeling/preference_model.py:146]: The original docstring was within the 80chars limit, no?

Re: your earlier comment, I still think this method should not have a return value, but `_assert_slot_exists()` might be a better name then.

File: usersimcrs/items/item_collection.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 """Represents a collection of items.
 
-Items are characterized by a set of properties, which correspond to slots in a
-domain. Items and can be loaded from CSV by providing a mapping from CSV fields
-to properties (i.e., domain slots).
+Items are characterized by a set of properties, which correspond to
+slots in a domain. Items and can be loaded from CSV by providing a
+mapping from CSV fields to properties (i.e., domain slots).
```
Comment (kbalog) [inline usersimcrs/items/item_collection.py:5]: Did something happen with the black configuration? The original text was within the 80 chars limit.

File: usersimcrs/items/item_collection.py [hunk 3]
```diff
@@ -113,5 +113,12 @@ def get_possible_property_values(self, property: str) -> List[str]:
         Returns:
             List of possible values.
         """
-        # TODO: https://github.com/iai-group/UserSimCRS/issues/112
-        return []
+        values: List[Any] = []
```
Comment (kbalog) [inline usersimcrs/items/item_collection.py:116]: Wouldn't it be more efficient to make it a set?

File: tests/items/test_item_collection.py [hunk 4]
```diff
@@ -57,3 +57,36 @@ def test_load_items_csv(domain: Domain, movie: Dict[str, Any]) -> None:
...
+    }
+
+    item_collection.load_items_csv(
+        ITEMS_CSV_FILE,
+        id_col="movieId",
+        domain=domain,
+        domain_mapping=mapping,
+    )
+
+    genres = item_collection.get_possible_property_values("GENRE")
```
Comment (kbalog) [inline tests/items/test_item_collection.py:81]: Also check for a value that's not supposed to be in a list of genres?

File: tests/items/test_item_collection.py [hunk 5]
```diff
@@ -57,3 +57,36 @@ def test_load_items_csv(domain: Domain, movie: Dict[str, Any]) -> None:
...
+    assert len(genres) == 20
+    assert {"Adventure", "Animation", "Children", "Comedy", "Fantasy"}.issubset(
+        genres
+    )
+
+    titles = item_collection.get_possible_property_values("TITLE")
+    assert len(titles) == 13813
+    assert "Toy Story (1995)" in titles
```
Comment (kbalog) [inline tests/items/test_item_collection.py:89]: Also check for a title that's not in the collection?

**Review by kbalog — CHANGES_REQUESTED** [review]

File: usersimcrs/user_modeling/preference_model.py (hunk 1 above)
Comment (NoB0) [reply usersimcrs/user_modeling/preference_model.py:146]: The docstring was 81 characters, so I ran docformatter to fix that.

I like to new name for the method.

File: usersimcrs/items/item_collection.py (hunk 2 above)
Comment (NoB0) [reply usersimcrs/items/item_collection.py:5]: I did not change the configuration, so not sure what happened here. I reverted these changes.

File: usersimcrs/items/item_collection.py (hunk 3 above)
Comment (NoB0) [reply usersimcrs/items/item_collection.py:116]: I was getting some error in another part of the code when I used a set. However, I found a workaround with a set.

File: tests/items/test_item_collection.py (hunk 4 above)
Comment (NoB0) [reply tests/items/test_item_collection.py:81]: Done.

**Review by kbalog — APPROVED** [review]
LGTM

**Pull request merged**